catkin_tools/verbs/catkin_build/build.py

```diff
@@ -461,49 +462,53 @@ def build_isolated_workspace(
+                continue
+
+            # Get actual execution deps
+            deps = [
+                p.name for _, p
+                in get_cached_recursive_build_depends_in_workspace(pkg, packages_to_be_built)
+                if p.name not in prebuild_jobs
+            ]
+            # All jobs depend on the prebuild jobs if they're defined
+            if not no_deps:
+                for j in prebuild_jobs.values():
+                    deps.append(j.jid)
+
+            # Determine the job parameters
+            build_job_kwargs = dict(
+                context=context,
+                package=pkg,
+                package_path=pkg_path,
+                dependencies=deps,
+                force_cmake=force_cmake,
+                pre_clean=pre_clean)
 
-        # Ignore metapackages
-        if 'metapackage' in [e.tagname for e in pkg.exports]:
-            continue
+            # Create the job based on the build type
+            build_type = get_build_type(pkg)
 
-        # Get actual execution deps
-        deps = [
-            p.name for _, p
-            in get_cached_recursive_build_depends_in_workspace(pkg, packages_to_be_built)
-            if p.name not in prebuild_jobs
-        ]
-        # All jobs depend on the prebuild jobs if they're defined
-        if not no_deps:
-            for j in prebuild_jobs.values():
-                deps.append(j.jid)
-
-        # Determine the job parameters
-        build_job_kwargs = dict(
-            context=context,
-            package=pkg,
-            package_path=pkg_path,
-            dependencies=deps,
-            force_cmake=force_cmake,
-            pre_clean=pre_clean)
-
-        # Create the job based on the build type
-        build_type = get_build_type(pkg)
-
-        if build_type in build_job_creators:
-            jobs.append(build_job_creators[build_type](**build_job_kwargs))
-        else:
-            wide_log(clr(
-                "[build] @!@{yf}Warning:@| Skipping package `{}` because it "
-                "has an unsupported package build type: `{}`"
-            ).format(pkg.name, build_type))
-
-            wide_log(clr("[build] Note: Available build types:"))
-            for bt_name in build_job_creators.keys():
-                wide_log(clr("[build]  - `{}`".format(bt_name)))
+            if build_type in build_job_creators:
+                jobs.append(build_job_creators[build_type](**build_job_kwargs))
+            else:
+                wide_log(clr(
+                    "[build] @!@{yf}Warning:@| Skipping package `{}` because it "
+                    "has an unsupported package build type: `{}`"
+                ).format(pkg.name, build_type))
+
+                wide_log(clr("[build] Note: Available build types:"))
+                for bt_name in build_job_creators.keys():
+                    wide_log(clr("[build]  - `{}`".format(bt_name)))
+
+    except CommandMissing as e:
+        sys.exit(clr("[build] @!@{rf}Error:@| {0}").format(e))
```

Comment:
You mean moving it to `catkin_build/cli.py`? That can be done. It'll mean importing the excepting there instead of here, which probably doesn't really matter.

The main thing is not catching everything— we still want stack traces for unknown problems.